packages/builder-rsbuild/src/index.ts

```diff
@@ -93,10 +94,9 @@ export const start: RsbuildBuilder['start'] = async ({
       ...config,
       server: {
         ...config.server,
-        port: options.port,
-        host: 'localhost',
+        port: await getRandomPort(options.host),
+        host: options.host,
         htmlFallback: false,
-        strictPort: true,
```

Comment:
This problem is caused by configuring `strictPort: true`. Rsbuild will parse the configuration (including calculating available port) when the server is initialized. When the port is occupied and `strictPort` is configured, an error will be reported.


<img width="1075" alt="image" src="https://github.com/user-attachments/assets/1f3b8df8-4aa0-4f80-8bf1-772005afe8c2">


https://github.com/web-infra-dev/rsbuild/blob/v0.7.9/packages/core/src/server/helper.ts#L219-L224

Reply:
Got it, I'll remove port setting.